### PR TITLE
[Fix Bug] Realtime Node may double count query results when multiple spec files are specified.

### PR DESCRIPTION
**Pull request by guobingkun**
With the current implementation, if realtime node takes more than 1 spec, there are 2 cases I can think of that could cause problems,

- you have two realtime nodes, one is a replicate, and there are two linear partitions on each node.  It's possible query that uses partition0 hits realtime1 while partition1 hits realtime2, which causes double counting on each realtime nodes. 

- you have 1 realtime node, and there are two linear partitions. It's possible that when query comes, partition0 is already handed off to historical while partition1 hasn't. Which ended up double counting on realtime node.

### CODE REVIEW

File: server/src/main/java/io/druid/segment/realtime/RealtimeManager.java [hunk 1]
```diff
@@ -139,49 +153,87 @@ public FireDepartmentMetrics getMetrics(String datasource)
...
+              FireChief retVal = partitionChiefs.get(spec.getPartitionNumber());
+              return retVal == null ? new NoopQueryRunner<T>() : retVal.getQueryRunner(query);
+            }
+          }
+      );
+      runners.add(
+          toolchest.mergeResults(factory.mergeRunners(MoreExecutors.sameThreadExecutor(), subRunners))
+      );
+    }
+    return new UnionQueryRunner<>(factory.mergeRunners(MoreExecutors.sameThreadExecutor(), runners));
```
Comment (himanshug) [inline server/src/main/java/io/druid/segment/realtime/RealtimeManager.java]: not sure if UnionQueryRunner is necessary, becoz that would be covered by the broker. Realtime node should never be receiving union queries.

Comment (guobingkun) [inline server/src/main/java/io/druid/segment/realtime/RealtimeManager.java]: I agree, @nishantmonu51 any suggestion?

Comment (nishantmonu51) [inline server/src/main/java/io/druid/segment/realtime/RealtimeManager.java]: thats true, realtime nodes dont get union query. they are broken down into separate queries at the broker.

**Comment by cheddar** [comment]
:+1:

File: server/src/main/java/io/druid/segment/realtime/RealtimeManager.java (hunk 1 above)
Comment (guobingkun) [inline server/src/main/java/io/druid/segment/realtime/RealtimeManager.java]: @nishantmonu51 What if a query's datasource is QueryDataSource, and its nested query's datasource is UnionDataSource?

Comment (guobingkun) [inline server/src/main/java/io/druid/segment/realtime/RealtimeManager.java]: nvm, it shouldn't happen on realtime. Updated PR, please check @himanshug @nishantmonu51

**Pull request merged**